Zend/zend_compile.c

```diff
@@ -5302,6 +5341,23 @@ static zend_type zend_compile_typename(zend_ast *ast, zend_bool force_allow_null
 				zend_string_addref(class_name);
 			}
 
+			if (ast->attr == ZEND_NAME_NOT_FQ
+					&& zend_is_confusable_type(orig_name, &correct_name)
+					&& zend_is_not_imported(orig_name)) {
+				if (correct_name) {
+					zend_error(E_COMPILE_WARNING,
+						"\"%s\" will be interpreted as a class type. Did you mean \"%s\"? "
```

Comment:
I find `class type` to be confusing, ironically. I'd suggest `class name` as the more idiomatic way to express this:

> "Integer" will be interpreted as a class name. Did you mean "int"?

Reply:
I am the only one for whom the future tense seems a bit odd? Wouldn't present tense sound more natural? Like below:

> Warning: "integer" is interpreted as a class type (name). Did you mean "int"?

Reply:
Not just you @kocsismate. I also thought this, but already had two grammar comments. Didn't want to push my luck!